hs/database/Tables.hs

```diff
@@ -65,6 +65,32 @@ Distribution
     dId Int
     description String
     deriving Show
+
+Graphs
+    gId Int
+    title String
+    deriving Show
+
+Rects
+    gId Int
+    rId Int
+    width Rational
+    height Rational
+    xPos Rational
+    yPos Rational
+    colour String
+    deriving Show
+
+Texts
+    gId Int
+    rId Int
+    yPos Rational
```

Comment:
Hmmm. Text which belongs to a rect should be centred inside the rect.

Text that doesn't belong to a rect (region labels?) need to have both x and y positions.

All three of these attributes (rId, yPos, xPos) should be Maybes.